aidants_connect_web/admin.py

```diff
@@ -22,38 +23,71 @@
 admin_site = OTPAdminSite(OTPAdminSite.name)
 
 
-class AidantAdmin(DjangoUserAdmin):
-    # The forms to add and change aidant instances
+class VisibleToStaff(ModelAdmin):
+    """A mixin to make a ModelAdmin visible to staff users."""
+
+    def has_module_permission(self, request):
+        return request.user.is_staff
+
+    def has_view_permission(self, request, obj=None):
+        return self.has_module_permission(request)
+
+    def has_add_permission(self, request, obj=None):
+        return self.has_module_permission(request)
+
+    def has_change_permission(self, request, obj=None):
+        return self.has_module_permission(request)
+
+    def has_delete_permission(self, request, obj=None):
+        return self.has_module_permission(request)
+
+
+class StaffStaticDeviceAdmin(VisibleToStaff, StaticDeviceAdmin):
+    pass
+
+
+class StaffTOTPDeviceAdmin(VisibleToStaff, TOTPDeviceAdmin):
+    pass
+
+
+class AidantAdmin(VisibleToStaff, DjangoUserAdmin):
+
+    def get_form(self, request, obj=None, **kwargs):
+        form = super().get_form(request, obj, **kwargs)
+
+        # Prevent non-superusers from being able to set
+        # the `is_staff` and `is_superuser` flags.
+        if not request.user.is_superuser:
+            disabled_fields = {
+                'is_staff',
+                'is_superuser',
+            }
+            for f in disabled_fields:
+                if f in form.base_fields:
+                    form.base_fields[f].disabled = True
```

Comment:
Pour l'amour de supprimer des lignes de code
```suggestion
        if not request.user.is_superuser:
                if "is_superuser" in form.base_fields:
                    form.base_fields["is_superuser"].disabled = True
                if "is_staff" in form.base_fields:
                    form.base_fields["is_staff"].disabled = True                
```